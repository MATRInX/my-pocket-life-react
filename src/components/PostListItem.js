import React from 'react';
import moment from 'moment';
import { Link } from 'react-router-dom';
import { CATEGORIES } from '../fixtures/links';

moment.locale('pl');

export const PostListItem = ({id, image, url, published, title, labels, content}) => {
  const publishedFromNow = moment(published).fromNow();
  const shortUrl = `/post${url.substr(url.lastIndexOf('/'))}`;
  const labelComponent = labels.map((label, index) => {
    for (const prop in CATEGORIES) {
      if(label === CATEGORIES[prop].name) {
        return (
          <span key={index}>
            <Link to={CATEGORIES[prop].link}>
              {CATEGORIES[prop].name}
            </Link>
            {index < labels.length-1 ? ', ': ''}
          </span>
        );
      };
    };
  });
  return (
    <div>
      <Link to={shortUrl}>
        <h2>{title}</h2>
      </Link>
      <p>{publishedFromNow}</p>
      <Link to={shortUrl}>
        <img src={image} />
      </Link>
      <p>{content}</p>
      {labelComponent}
    </div>
  )
};

export default PostListItem;