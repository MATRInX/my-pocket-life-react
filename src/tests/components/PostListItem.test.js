import React from 'react';
import { shallow } from 'enzyme';
import PostListItem from '../../components/PostListItem';
import { dummyPosts as posts } from '../../fixtures/dummyData';

test('should correctly render PostListItem', () => {
  const wrapper = shallow(<PostListItem {...posts[0]}/>);
  expect(wrapper).toMatchSnapshot();
});