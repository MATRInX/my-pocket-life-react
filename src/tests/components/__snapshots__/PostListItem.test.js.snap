// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render PostListItem 1`] = `
<div>
  <Link
    replace={false}
    to="/post/latest-updates-august-1st.html"
  >
    <h2>
      Sample title1
    </h2>
  </Link>
  <p>
    7 lat temu
  </p>
  <Link
    replace={false}
    to="/post/latest-updates-august-1st.html"
  >
    <img />
  </Link>
  <p>
    My short content1 to show on main page
  </p>
  <span
    key="0"
  >
    <Link
      replace={false}
      to="/kategoria/diy"
    >
      DIY
    </Link>
    , 
  </span>
  <span
    key="1"
  >
    <Link
      replace={false}
      to="/kategoria/news"
    >
      News
    </Link>
  </span>
</div>
`;
